src/sentry/toolbar/iframe_view.py

```diff
@@ -0,0 +1,40 @@
+from typing import Any
+
+from django.http import HttpRequest, HttpResponse
+
+from sentry.web.frontend.base import OrganizationView, region_silo_view
+
+
+@region_silo_view
+class IframeView(OrganizationView):
+    # TODO: For perms check. This is just an example and we might not need it.
+    # required_scope = "org:read,org:integrations"
+
+    security_headers = {
+        "X-Frame-Options": "ALLOWALL",
+        # TODO: # Not working, seems to get overridden by a BaseView wrapper
+        # "Content-Security-Policy": "frame-ancestors http://dev.getsentry.net/ http: https:",
+    }
+
+    def respond(self, template: str, context: dict[str, Any] | None = None, status: int = 200):
+        response = super().respond(template, context=context, status=status)
+        for header, val in IframeView.security_headers.items():
+            response[header] = val
+        return response
+
+    def handle_auth_required(self, request: HttpRequest, *args, **kwargs):
+        # Override redirects to login
+        if request.method == "GET":
+            self.default_context = {}
+            return self.respond("sentry/toolbar/iframe-no-auth.html", status=401)
```

Comment:
lets return the same `iframe.html` template in all cases for now. We have the python code here to render different templates which is great, and when we have different content we can make the new files.

One thing that might happen is... we have html content that's very similar in each case, and instead of having a new template each time we can use template variables to tweak the shared template.